### PR TITLE
refactor: build a bull set no snapshots

### PR DESCRIPTION
**Pull request by PhearZero**
# Overview

- Refactors build-a-bull from `PARTITIONED_WEIGHTING` to `NO_SNAPSHOT`

## Issues:

- Tracking: https://algorandfoundation.atlassian.net/browse/ENG-67

### CODE REVIEW

**Comment by PhearZero** [comment]
@ori-shem-tov if you have time, give this a spin. It's configured for `NO_SNAPSHOT` and it uses the same `csv` as xgov but ignores the weights.

File: src/build-a-bull/src/features/vote-creation/review/index.tsx [hunk 1]
```diff
@@ -56,12 +56,34 @@ export default function Review() {
         setAuth(await auth.execute({ signer }))
         break
       case VoteCreationReviewSteps.Create:
+        console.log({
```
Comment (ori-shem-tov) [inline src/build-a-bull/src/features/vote-creation/review/index.tsx:59]: debugging?

File: src/build-a-bull/src/features/vote-creation/review/index.tsx [hunk 2]
```diff
@@ -73,7 +95,7 @@ export default function Review() {
                   threshold: Number(proposal.threshold),
                   ask: Number(proposal.ask),
                 },
-                answers: ['yes'],
+                answers: ['Yes', 'No'],
```
Comment (ori-shem-tov) [inline src/build-a-bull/src/features/vote-creation/review/index.tsx:98]: what's the purpose of adding `No`?

File: src/build-a-bull/src/features/vote-creation/review/index.tsx [hunk 3]
```diff
@@ -89,10 +111,12 @@ export default function Review() {
 
         break
       case VoteCreationReviewSteps.Bootstrap:
+        console.log(proposals)
```
Comment (ori-shem-tov) [inline src/build-a-bull/src/features/vote-creation/review/index.tsx:114]: debugging 2

File: src/build-a-bull/src/features/vote/index.tsx [hunk 4]
```diff
@@ -247,35 +207,14 @@ function Vote({ sort: sortProp = 'none' }: { sort?: 'ascending' | 'descending' |
...
-        })
-      }
-    }
-
     await submitVote({
-      signature: allowlistSignature,
-      selectedOptionIndexes: votingRoundMetadata.questions.map(() => 0),
-      weighting: voteWeight,
-      weightings: votingRoundMetadata.questions.map((question) => (newVoteAllocations[question.id] ? newVoteAllocations[question.id] : 0)),
+      signature: allowlistSignature || '',
```
Comment (ori-shem-tov) [inline src/build-a-bull/src/features/vote/index.tsx:211]: why not just `''`?

File: src/build-a-bull/src/features/vote/index.tsx [hunk 5]
```diff
@@ -77,16 +75,18 @@ function Vote({ sort: sortProp = 'none' }: { sort?: 'ascending' | 'descending' |
   const hasVoteEnded = !votingRoundGlobalState ? false : getHasVoteEnded(votingRoundGlobalState)
   const isVoteCreator = votingRoundMetadata?.created.by === activeAddress ? true : false
 
-  const canVote = hasVoteStarted && !hasVoteEnded && allowedToVote
+  const canVote = hasVoteStarted && !hasVoteEnded
   const hasVoted = voterVotes !== undefined ? true : false
   const hasClosed = votingRoundGlobalState && votingRoundGlobalState.close_time !== undefined ? true : false
 
+
+  const totalVotes = useMemo(()=>votingRoundResults ? votingRoundResults.reduce((c, r)=>c + r.count, 0) : 0, [votingRoundResults])
```
Comment (ori-shem-tov) [inline src/build-a-bull/src/features/vote/index.tsx:83]: did you make sure that this works well now that you have 2 options per question?

File: src/build-a-bull/src/root.tsx [hunk 6]
```diff
@@ -54,7 +54,7 @@ export default function Root() {
           }
         />
         <SiteFooter />
-        <ScrollToTop offset={offset} />
+        {/*<ScrollToTop offset={offset} />*/}
```
Comment (ori-shem-tov) [inline src/build-a-bull/src/root.tsx:57]: ?

File: src/build-a-bull/src/features/vote-creation/review/index.tsx (hunk 2 above)
Comment (PhearZero) [reply src/build-a-bull/src/features/vote-creation/review/index.tsx:98]: The original `dapp` had a requirement for a minimum of 2 options. Will test without it once we are staged

File: src/build-a-bull/src/features/vote/index.tsx (hunk 5 above)
Comment (PhearZero) [reply src/build-a-bull/src/features/vote/index.tsx:83]: Just saw the same bug locally, need to filter for Yes options

**Pull request merged**